README.md

```diff
@@ -48,32 +41,123 @@ So that I am reassured that my order will be delivered on time
-  * The text sending functionality should be implemented using Twilio API. You'll need to register for it. It’s free.
-  * Use the twilio-ruby gem to access the API
-  * Use the Gemfile to manage your gems
-  * Make sure that your Takeaway is thoroughly tested and that you use mocks and/or stubs, as necessary to not to send texts when your tests are run
-  * However, if your Takeaway is loaded into IRB and the order is placed, the text should actually be sent
-  * Note that you can only send texts in the same country as you have your account. I.e. if you have a UK account you can only send to UK numbers.
+Here is an example of a feature test of my program:
+```
+2.4.0 :001 > require './lib/order.rb'
+ => true
+2.4.0 :002 > menu = Menu.new
+ => #<Menu:0x007fbf5e036e78>
+2.4.0 :003 > order = Order.new
+ => #<Order:0x007fbf5e1a1c90 @basket=[], @updated_basket={}>
+2.4.0 :004 > order.order_dish('Kobe Slider')
+1 Kobe Slider(s) added to your basket.
+ => nil
+2.4.0 :005 > order.order_dish(3, 'Shrimp Tempura')
+3 Shrimp Tempura(s) added to your basket.
+ => nil
+2.4.0 :006 > order.show_basket
+(1) Kobe Slider(s): £7
+(3) Shrimp Tempura(s): £12
+Total to pay: £19
+ => nil
+2.4.0 :007 > order.place_order
+```
+
+Approach to solving the challenge:
+----------------------------------
+
+First, I created a domain model for each User Story (above) to keep track of the specified criteria.
+
+My program ensures that:
 
-* Advanced! (have a go if you're feeling adventurous):
-  * Implement the ability to place orders via text message.
+* The menu can be read; it gives the list of dishes and associated prices
+* The customer can choose a quantity for the available dishes from the menu
+* If the dish is not listed on the menu, the customer is prompted to select an available item from the menu
+* The customer can see their basket which gives the total items and prices of the various dishes they ordered
+* The customer receives a text when their order is placed and what time to expect their delivery to arrive
 
-* A free account on Twilio will only allow you to send texts to "verified" numbers. Use your mobile phone number, don't worry about the customer's mobile phone.
-* Finally submit a pull request before Monday at 9am with your solution or partial solution.  However much or little amount of code you wrote please please please submit a pull request before Monday at 9am
+Creating the menu
+-----------------
 
+I wrote a few test cases for creating my menu. First describing what I wanted to test and then running it through RSPEC and expecting errors to occur.
 
-In code review we'll be hoping to see:
+```ruby
+describe Menu do
+  describe '#initialize' do
+    it { is_expected.to respond_to(:initialize) }
+  end
+```
+
+I expected the errors to show that I didn't have a 'Menu' class or an 'initialize' method. So, then proceeded to write the code for the tests to pass.
```

Comment:
These process analyses are really great! It's worth trying to critique your process.  Are they any approaches you took that you realise you could do better next time?